File: androidweekview/src/main/java/com/alamkanak/weekview/TopLeftCornerDrawer.java
package com.alamkanak.weekview;

import android.graphics.Canvas;
import android.graphics.Rect;

public class TopLeftCornerDrawer {
	private WeekViewConfig config;
	private WeekViewDrawingConfig drawConfig;

	TopLeftCornerDrawer(WeekViewConfig config) {
		this.config = config;
		this.drawConfig = config.drawingConfig;
	}

	void draw(Canvas canvas) {
		final int headerHeight = (int) (drawConfig.headerHeight
				+ config.headerRowPadding * 2
				+ drawConfig.headerMarginBottom);
		final int timeColumnWidth = (int) drawConfig.timeColumnWidth;

		final Rect sourceRect = new Rect(0, 0, config.topLeftCornerDrawable.getIntrinsicWidth(), config.topLeftCornerDrawable.getIntrinsicHeight());
		final Rect destRect = new Rect(0, 0, timeColumnWidth, headerHeight);
		Rect bounds;

		if (destRect.height() >= destRect.width()) {
			int diff = (destRect.height() - sourceRect.height()) / 4;
			bounds = new Rect(config.topLeftCornerPadding, diff + config.topLeftCornerPadding,
					destRect.right - config.topLeftCornerPadding, destRect.bottom - diff - config.topLeftCornerPadding);
		} else {
			int diff = (destRect.width() - sourceRect.width()) / 4;
			bounds = new Rect(diff + config.topLeftCornerPadding, config.topLeftCornerPadding,
					destRect.right - diff + config.topLeftCornerPadding, destRect.bottom - config.topLeftCornerPadding);
		}

		config.topLeftCornerDrawable.setBounds(bounds);
		config.topLeftCornerDrawable.draw(canvas);
	}
}

File: androidweekview/src/main/java/com/alamkanak/weekview/drawers/TopLeftCornerDrawer.kt
package com.alamkanak.weekview.drawers

import android.graphics.Canvas
import android.graphics.Rect
import com.alamkanak.weekview.DrawingContext
import com.alamkanak.weekview.config.WeekViewConfig
import com.alamkanak.weekview.config.WeekViewDrawConfig

class TopLeftCornerDrawer(private val config: WeekViewConfig) : BaseDrawer {
	private val drawConfig: WeekViewDrawConfig = config.drawConfig

	override fun draw(drawingContext: DrawingContext, canvas: Canvas) {
		config.topLeftCornerDrawable?.let { drawable ->
			val timeColumnWidth = drawConfig.timeColumnWidth.toInt()

			val destRect = Rect(0, 0, timeColumnWidth, drawConfig.headerHeight.toInt())

			val imageSize = destRect.shortestSide()
			val hDiff = destRect.width() - imageSize
			val vDiff = destRect.height() - imageSize

			drawable.bounds = Rect(
					hDiff / 2 + config.topLeftCornerPadding,
					vDiff / 2 + config.topLeftCornerPadding,
					hDiff / 2 + imageSize - config.topLeftCornerPadding,
					vDiff / 2 + imageSize - config.topLeftCornerPadding
			)

			drawable.draw(canvas)
		}
	}

	private fun Rect.shortestSide(): Int {
		return Math.min(width(), height())
	}
}